libraries/SuplaDevice/supla-common/srpc.c

```diff
@@ -1409,7 +1409,8 @@ _supla_int_t SRPC_ICACHE_FLASH srpc_ds_async_registerdevice(
       (sizeof(TDS_SuplaDeviceChannel) * SUPLA_CHANNELMAXCOUNT) +
       (sizeof(TDS_SuplaDeviceChannel) * registerdevice->channel_count);
 
-  if (size > sizeof(TDS_SuplaRegisterDevice)) return 0;
+  if (size > sizeof(TDS_SuplaRegisterDevice)) return SUPLA_RESULT_DATA_TOO_LARGE;
+  if (size > SUPLA_MAX_DATA_SIZE) return SUPLA_RESULT_DATA_TOO_LARGE;
```

Comment:
Such verification should be common to all calls. It's best if you put in some callback for errors here
https://github.com/SUPLA/supla-core/blob/develop/supla-common/srpc.c#L227. Such a callback would have to be behind some ifdef.

#ifdef ERROR_CALLBACK
if (srpc->params.on_error) {
srpc->params.on_error(...)
}
#endif

https://github.com/SUPLA/supla-core/blob/develop/supla-common/srpc.h#L77

#ifdef ERROR_CALLBACK
_func_srpc_OnError on_error;
#endif